util/expensivequery/expensivequery.go

```diff
@@ -73,12 +85,214 @@ func (eqh *Handle) Run() {
 				}
 			}
 			threshold = atomic.LoadUint64(&variable.ExpensiveQueryTimeThreshold)
+			if eqh.record.err == nil {
+				eqh.oomKillerAlert()
+			}
 		case <-eqh.exitCh:
 			return
 		}
 	}
 }
 
+type memoryUsageAlarmRecord struct {
+	err               error
+	useInstanceMemory bool // useInstanceMemory indicate whether use instance memory or system memory to alarm oom risk.
+	serverMemoryQuota uint64
+	lastRecordTime    time.Time
+
+	tmpDir              string
+	lastLogFileName     []string
+	lastProfileFileName [][]string // heap, goroutine
+}
+
+func (eqh *Handle) initMemoryUsageAlarmRecord() {
+	if config.GetGlobalConfig().Performance.ServerMemoryQuota != 0 {
+		eqh.record.serverMemoryQuota = config.GetGlobalConfig().Performance.ServerMemoryQuota
+		eqh.record.useInstanceMemory = true
+	} else {
+		eqh.record.serverMemoryQuota, eqh.record.err = memory.MemTotal()
+		if eqh.record.err != nil {
+			logutil.BgLogger().Error("Get system total memory fail.", zap.Error(eqh.record.err))
+			return
+		}
+		eqh.record.useInstanceMemory = false
+	}
+	eqh.record.lastRecordTime = time.Time{}
+	eqh.record.tmpDir = config.GetGlobalConfig().TempStoragePath
+	if alert := config.GetGlobalConfig().Performance.MemoryUsageAlarmRatio; alert == 0 || alert == 1 {
+		eqh.record.err = errors.New("close memory usage alarm recorder")
+	}
+	eqh.record.lastProfileFileName = make([][]string, 2)
+}
+
+// If Performance.ServerMemoryQuota is set, use instance memory usage and ServerMemoryQuota * 80% to check oom risk.
+// If Performance.ServerMemoryQuota is not set, use system memory usage and total memory * 80% to check oom risk.
```

Comment:
use system memory usage and total memory * 80% to check oom risk

--> 
use system total memory usage * 80% to check oom risk